src/v/cluster/health_monitor_backend.cc

```diff
@@ -151,16 +159,62 @@ std::optional<node_health_report> health_monitor_backend::build_node_report(
     return report;
 }
 
+void health_monitor_backend::abortable_refresh_request::abort() {
+    if (finished) {
+        return;
+    }
+    finished = true;
+    done.set_value(errc::timeout);
+}
+
+health_monitor_backend::abortable_refresh_request::abortable_refresh_request(
+  model::node_id leader_id, ss::gate::holder holder, ss::semaphore_units<> u)
+  : leader_id(leader_id)
+  , holder(std::move(holder))
+  , units(std::move(u)) {}
+
+void health_monitor_backend::abortable_refresh_request::set_fiber(
+  ss::future<std::error_code> f) {
+    (void)std::move(f).then_wrapped(
+      [self = shared_from_this()](ss::future<std::error_code> f) {
+          if (self->finished) {
+              return;
+          }
+          self->finished = true;
+          if (f.failed()) {
+              self->done.set_exception(f.get_exception());
+          } else {
+              self->done.set_value(f.get());
+          }
+      });
+}
+
+ss::future<std::error_code>
+health_monitor_backend::abortable_refresh_request::when_done() {
```

Comment:
I found the `when_done` name a bit confusing, maybe `await` instead?  The `done` variable name makes sense inside the class but isn't otherwise part of the interface.